sklearn/decomposition/_nmf.py

```diff
@@ -976,9 +979,15 @@ def non_negative_factorization(
         If `update_H=False`, it is used as a constant, to solve for W only.
         If `None`, uses the initialisation method specified in `init`.
 
-    n_components : int, default=None
+    n_components : int or {'auto'} or None, default=None
         Number of components, if n_components is not set all features
         are kept.
+        If `n_components='auto'`, the number of components is automatically infered
+        from W or H shapes.
```

Comment:
```suggestion
        If `n_components='auto'`, the number of components is automatically inferred
        from W or H shapes.
```

Reply:
fixed in b3529bb1a0b276990b2124907d9b2f9b8f790b5d